File: 011-020/011/011.py
#!/usr/bin/env python

grid = []

with open("grid", "r") as mygrid:
    mylines = mygrid.readlines()
    for gridn in enumerate(mylines):
        grid.append(mylines[gridn[0]].split())

biggest = 0

for row in range(0, len(grid)):
    for column in range(0, len(grid[row])):
        mynumber = int(grid[row][column])
        total = [mynumber]*4
        for x in range(1, 4):
            # Vertical
            try:
                total[0] = total[0] * int(grid[row + x][column])
            except IndexError:
                total[0] = 1
            # Horizontal
            try:
                total[2] *= int(grid[row][column + x])
            except IndexError:
                total[2] = 1
            # Diagonal Right
            try:
                total[1] *= int(grid[row + x][column + x])
            except IndexError:
                total[1] = 1
            # Diagonal Left
            try:
                total[3] *= int(grid[row - x][column + x])
            except IndexError:
                total[3] = 1
        if max(total) > biggest:
            biggest = max(total)

print(biggest)

File: 011-020/011/011.rb
#!/usr/local/bin/ruby -w

grid = []

File.open("grid", "r") do |mygrid|
    i = 0
    mygrid.each_line do |line|
        grid[i] = line.split
        i += 1
    end
end

biggest = 0

(0..grid.length - 1).each do |row|
    (0..grid[row].length - 1).each do |column|
        mynumber = grid[row][column].to_i
        total = [mynumber] * 4
        (1..3).each do |x|
            # Vertical
            begin
                total[0] *= grid[row + x][column].to_i
            rescue
                total[0] = 1
            end
            # Horizontal
            begin
                total[2] *= grid[row][column + x].to_i
            rescue
                total[2] = 1
            end
            # Diagonal Right
            begin
                total[1] *= grid[row + x][column + x].to_i
            rescue
                total[1] = 1
            end
            # Diagonal Left
            begin
                total[3] *= grid[row - x][column + x].to_i
            rescue
                total[3] = 1
            end
        end

        biggest = total.max if total.max > biggest
    end
end

puts biggest
